### ISSUE TITLE
Webapi: Configurable product can not be updated

### ISSUE THREAD

**Issue by pmzandbergen** [magento/magento2#35907]
### Preconditions and environment

Magento version `>= 2.4.4-p1`

### Steps to reproduce

1. Create a Configurable Product with 2 simples (configurations), called option A and B. 
2. Create a Cart containing the Configurable Product using option A
3. Update the Cart Item to use option B using the REST API without updating the quantity (`PUT guest-carts/{cartId}/items/{itemId}`) *
4. The Cart Item is still using option A

_* Example PUT content_
```
{
   "cartItem":{
      "quote_id":1,
      "qty":1,
      "product_option":{
         "extension_attributes":{
            "configurable_item_options":[
               {
                  "option_id":"1",
                  "option_value":"2"
               }
            ]
         }
      }
   }
}
```

### Expected result

The cart item is being updated

### Actual result

The cart item is unaffected

### Additional information

**Cause**
The BuyRequest is being filtered on updated quantity, see commit:
https://github.com/magento/magento2/commit/5478c03cfe34030ad377594c7e6398836d304477

**Suggested fix**
Rollback this commit, the original issue should be fixed some other way.

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by pmzandbergen** [comment]
@magento give me 2.4-develop instance

**Comment by pmzandbergen** [comment]
### Confirmed

**Create Cart**
_Request_
```
`curl --location --request POST 'https://81894b31c95832aae4b00b54d38377ac.instances.magento-community.engineering/rest/default/V1/guest-carts'`
```
_Response_
```
"VlDQgTJURRWXekuQFBfzMp2yOTmfMjar"
```

**Add Item**
_Request_
```
curl --location --request POST 'https://81894b31c95832aae4b00b54d38377ac.instances.magento-community.engineering/rest/default/V1/guest-carts/VlDQgTJURRWXekuQFBfzMp2yOTmfMjar/items' \
--header 'Content-Type: application/json' \
--data-raw '{
    "cartItem": {
        "quote_id": "IScEXP5mP7lACkEJOlfJrWgEQULSD3GP",
        "qty": "1.0",
        "sku": "MH01",
        "product_option": {
            "extension_attributes": {
                "configurable_item_options": [
                    {
                        "option_id": "93",
                        "option_value": "49"
                    },
                    {
                        "option_id": "142",
                        "option_value": "166"
                    }
                ]
            }
        }
    }
}'
```
_Response_
```
{
    "item_id": 13,
    "sku": "MH01-XS-Black",
    "qty": 1,
    "name": "Chaz Kangeroo Hoodie",
    "price": 52,
    "product_type": "configurable",
    "quote_id": "6",
    "product_option": {
        "extension_attributes": {
            "configurable_item_options": [
                {
                    "option_id": "93",
                    "option_value": 49
                },
                {
                    "option_id": "142",
                    "option_value": 166
                }
            ]
        }
    }
}
```

**Update Item**
_Request_
```
curl --location --request PUT 'https://81894b31c95832aae4b00b54d38377ac.instances.magento-community.engineering/rest/default/V1/guest-carts/VlDQgTJURRWXekuQFBfzMp2yOTmfMjar/items/13' \
--header 'Content-Type: application/json' \
--data-raw '{
    "cartItem": {
        "item_id": 7,
        "quote_id": "8awS3tgp0SE6afe4UDpRsQQm0vfd8s7B",
        "qty": "1.0",
        "sku": "MH01",
        "product_option": {
            "extension_attributes": {
                "configurable_item_options": [
                    {
                        "option_id": "93",
                        "option_value": "49"
                    },
                    {
                        "option_id": "142",
                        "option_value": "167"
                    }
                ]
            }
        }
    }
}'
```
_Response_
```
{
    "item_id": 13,
    "sku": "MH01-XS-Black",
    "qty": 1,
    "name": "Chaz Kangeroo Hoodie",
    "price": 52,
    "product_type": "configurable",
    "quote_id": "6",
    "product_option": {
        "extension_attributes": {
            "configurable_item_options": [
                {
                    "option_id": "93",
                    "option_value": 49
                },
                {
                    "option_id": "142",
                    "option_value": 166
                }
            ]
        }
    }
}
```
The product options should have been updated at this point, but they are not.
**Update Item including its quantity**
_Request_
```
curl --location --request PUT 'https://81894b31c95832aae4b00b54d38377ac.instances.magento-community.engineering/rest/default/V1/guest-carts/VlDQgTJURRWXekuQFBfzMp2yOTmfMjar/items/13' \
--header 'Content-Type: application/json' \
--data-raw '{
    "cartItem": {
        "item_id": 7,
        "quote_id": "8awS3tgp0SE6afe4UDpRsQQm0vfd8s7B",
        "qty": "2.0",
        "sku": "MH01",
        "product_option": {
            "extension_attributes": {
                "configurable_item_options": [
                    {
                        "option_id": "93",
                        "option_value": "49"
                    },
                    {
                        "option_id": "142",
                        "option_value": "167"
                    }
                ]
            }
        }
    }
}'
```
_Response_
```
{
    "item_id": 15,
    "sku": "MH01-S-Black",
    "qty": 2,
    "name": "Chaz Kangeroo Hoodie",
    "price": 52,
    "product_type": "configurable",
    "quote_id": "6",
    "product_option": {
        "extension_attributes": {
            "configurable_item_options": [
                {
                    "option_id": "93",
                    "option_value": 49
                },
                {
                    "option_id": "142",
                    "option_value": 167
                }
            ]
        }
    }
}
```
Both quantity and options have been updated.

**Comment by simsComputing** [comment]
@magento I am working on this

**Comment by simsComputing** [comment]
@magento add to contributors team

**Comment by simsComputing** [comment]
@magento I am working on this

**Comment by pmzandbergen** [comment]
@simsComputing has already been fixed, see https://github.com/magento/magento2/pull/35908#issuecomment-1243411000

**Comment by simsComputing** [comment]
@pmzandbergen , issue was fixed in commit eb02e71
i think this issue can be closed as is. I don't know the procedure.

**Comment by simsComputing** [comment]
Wow i didn't see your previous messages. Sorry.
